### PR TITLE
Redirect to original URL after login

### PR DESCRIPTION
**Pull request by illfixit**
_What issues does this PR close?_

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [ ] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: authority-portal-frontend/src/app/core/global-state/routes/url-before-login.service.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            event.url != '/random-redirect-for-force-refresh',
+        ),
+        first(),
+      )
+      .subscribe((event) => {
+        this.originalUrl = event.urlAfterRedirects || event.url;
+        localStorage.setItem('originalUrl', this.originalUrl);
+      });
+  }
+
```
Comment (illfixit) [inline authority-portal-frontend/src/app/core/global-state/routes/url-before-login.service.ts:30]:
```suggestion
int test = "hello Suggestion"
```

**Comment by richardtreier** [comment]
closed in favor of https://github.com/sovity/authority-portal/pull/247

File: authority-portal-frontend/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -93,13 +93,21 @@ const REDIRECT_TO_HOME: string[] = [
...
 }));
 
 export const HOME_REDIRECTS: Routes = REDIRECT_TO_HOME.map((path) => ({
   path,
-  redirectTo: 'home',
+  redirectTo: (() => {
+    const url = localStorage.getItem('originalUrl') || 'home';
+    localStorage.removeItem('originalUrl');
+    return url;
+  })(),
```
Comment (kamilczaja) [inline authority-portal-frontend/src/app/app-routing.module.ts:110]: DRY

**Review by kamilczaja — CHANGES_REQUESTED** [review]

**Review by kamilczaja — APPROVED** [review]
LGTM

**Pull request merged**